### ISSUE TITLE
Error compiling to JavaScript. XMLHttpRequest error.

### ISSUE THREAD

**Issue by swavkulinski** [dart-lang/dart-pad#1554]
## What happened?

Console: Error compiling to JavaScript:
XMLHttpRequest error.

It was working fine through the morning. Suddenly it started to fail on every request. Is backend down?

## Steps to reproduce problem

Start new Pad with Dart only
Press start button
Nothing happens
Try reload the page

error shows in the console:
Console: Error compiling to JavaScript:
XMLHttpRequest error.

### Browser

Browser: Chrome

Version: Version 81.0.4044.138 (Official Build) (64-bit) (MacOSX)

Are you using any extensions/plugins that affect website behavior
(particularly those that affect iframes, such as ad blockers)?

Are there any warnings or errors in your browser's JavaScript console?
If so, paste them below:

```
[paste warnings or errors here]
```

### Machine

Operating system: MacOSX

Version: Catalina

### Your code

What code was in the editor, if any, when the failure occurred? You
can paste it in below:

```
void main() {
  for (int i = 0; i < 5; i++) {
    print('hello ${i + 1}');
  }
}
```

### DartPad's output

Did DartPad print anything to the console pane? If so, paste it below:

```
Error compiling to JavaScript:
XMLHttpRequest error.
```

If you were running Flutter code, you can also paste an image of the
Flutter output directly into this bug report.

**Comment by Dalamar** [comment]
Same issue here.

**Comment by swavkulinski** [comment]
same on other browsers

**Comment by joshuadeguzman** [comment]
Encountered the same, I actually just filed #1556. Thought opting out to code blocks would solve the issue.

**Comment by camillo777** [comment]
Same here. I can add this error:

```
Error compiling to JavaScript:
TimeoutException after 0:01:00.000000: Future not completed
```

**Comment by prabakaran-Code** [comment]
Yes same Error found

**Comment by nagarjunvv** [comment]
Even I am facing same issue, it was working fine yesterday night.

**Comment by Eyitemi01** [comment]
same error over here

**Comment by kolmakovruslan** [comment]
Chrome console output
`
DevTools failed to load SourceMap: Could not load content for https://dartpad.dev/split.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://dartpad.dev/scripts/playground.dart.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
flutter:1 Access to XMLHttpRequest at 'https://dart-services.appspot.com/api/dartservices/v2/version' from origin 'https://dartpad.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
playground.dart.js:9292 POST https://dart-services.appspot.com/api/dartservices/v2/version net::ERR_FAILED
b4 @ playground.dart.js:9292
Cc @ playground.dart.js:1347
(anonymous) @ playground.dart.js:13182
(anonymous) @ playground.dart.js:1415
$2 @ playground.dart.js:6575
$1 @ playground.dart.js:6567
fl @ playground.dart.js:7371
$0 @ playground.dart.js:6933
eO @ playground.dart.js:1495
d0 @ playground.dart.js:6841
$0 @ playground.dart.js:6896
EF @ playground.dart.js:1525
EL @ playground.dart.js:1527
$1 @ playground.dart.js:6522
Fw @ playground.dart.js:532
(anonymous) @ playground.dart.js:540
childList (async)
$1 @ playground.dart.js:6529
zP @ playground.dart.js:1532
e9 @ playground.dart.js:1593
ap @ playground.dart.js:6853
cD @ playground.dart.js:11673
hJ @ playground.dart.js:11206
eval @ VM790:3
$0 @ playground.dart.js:6733
$1 @ playground.dart.js:6741
cN @ playground.dart.js:7350
$1 @ playground.dart.js:7387
CH @ playground.dart.js:1451
CI @ playground.dart.js:1445
jj @ playground.dart.js:11204
(anonymous) @ playground.dart.js:11782
(anonymous) @ playground.dart.js:1415
$2 @ playground.dart.js:6575
by @ playground.dart.js:1401
d3 @ playground.dart.js:11784
jO @ playground.dart.js:11687
Dd @ playground.dart.js:4863
As @ playground.dart.js:5023
(anonymous) @ playground.dart.js:17220
(anonymous) @ playground.dart.js:17215
dartProgram @ playground.dart.js:17218
(anonymous) @ playground.dart.js:17220
flutter:1 Access to XMLHttpRequest at 'https://dart-services.appspot.com/api/dartservices/v2/analyze' from origin 'https://dartpad.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
playground.dart.js:9292 POST https://dart-services.appspot.com/api/dartservices/v2/analyze net::ERR_FAILED
`

**Comment by swavkulinski** [comment]
Can't replicate anymore. Looks like the backend is working again.

**Comment by Eyitemi01** [comment]
> same error over here

it works now

**Comment by RedBrogdon** [comment]
First off, thanks to @swavkulinski and the rest of you for reporting.

This morning, we experienced an outage in dart-services that we're still investigating. We've been able to bring the service back up, as you noticed, but the root cause has yet to be identified.

From here, we're going to:

* Determine exactly what caused dart-services to get into a non-working state, and
* Determine how we can get better monitoring/alerting in place. This could have been a thirty-minute problem rather than a six-hour problem, had the right people been notified at the right time. 

I'm closing this issue, since calls to dart-services are once again working. While I'm bummed that so many people have been inconvenienced, the fact that so many of you noticed is a good signal that we're building something people like and care about. Thanks again for reporting the problem!

**Comment by allentv** [comment]
The CORS issue is still present. Here the screenshot from today

<img width="1424" alt="Screenshot 2020-05-12 at 19 38 09" src="https://user-images.githubusercontent.com/6534992/81732243-1f4fad80-9488-11ea-90d5-e7dba584e13a.png">

**Comment by nuggetsnetwork** [comment]
getting the same issue

**Comment by vineetg3** [comment]
Same issue being reproduced.
I tried enabling CORS, disabling adblock , tried on Chrome and Firefox(latest versions) and even used a vpn.

**Comment by allentv** [comment]
I am getting `The server encountered a temporary error and could not complete your request. Please try again in 30 seconds.` when I try a POST request through Postman.

**Comment by allentv** [comment]
So I downloaded the `dart-services` repo on my machine and ran the service locally. I made a POST request via Postman to the `analyze` endpoint and can confirm that the CORS headers are available in the response.

<img width="962" alt="Screenshot 2020-05-12 at 21 49 00" src="https://user-images.githubusercontent.com/6534992/81743946-69418f00-949a-11ea-8434-21f47d9a862c.png">

The issue seems to be a problem with the server which seems to be working now.

**Comment by domesticmouse** [comment]
Yeah it went sideways again. Restarted everything, this time with a rolling restart after four hours policy

**Comment by MohamedAlsadig** [comment]
I learned from someone in an Arabic forum that the cause of this problem is to prevent access from this geographical location, try to log in with a Vpn.

**Comment by gppam** [comment]
Just experienced this error. Went to dartpad after the dart.dev examples didn't work (which has the same error as dartpad).

<img width="750" alt="Screen Shot 2020-11-07 at 5 39 52 PM" src="https://user-images.githubusercontent.com/14272390/98437874-e2a3fc80-2120-11eb-8f8a-8bdfd98291b6.png">

**Comment by domesticmouse** [comment]
Please re-try. We recycle servers every 1.5 hours (give or take) in an effort to deal with these issues.

**Comment by Bukharee** [comment]
Error compiling to JavaScript:
XMLHttpRequest error. i dont know why im burnt out !!!

**Comment by HicksonWei** [comment]
![image](https://user-images.githubusercontent.com/29944597/107316289-48fb3b00-6ad3-11eb-94d4-0842eb051cf7.png)

**Chrome DevTools Console Error:**
`Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://dartpad.dev') does not match the recipient window's origin ('null').`

**Comment by patzu** [comment]
same error happening here as well. Why? Why? Why?

**Comment by domesticmouse** [comment]
Please raise a new issue with steps to reproduce the error.

**Comment by rshash30** [comment]
You  Should run a VPN and it will work correctly

**Comment by domesticmouse** [comment]
Actually, these errors were due to unhappy servers. We have done a lot of work to improve the stability of the compilation service. Using VPN service probably wouldn't have helped in this instance.

Thanks!

**Comment by tunjineo** [comment]
Error compiling to JavaScript:
LateInitializationError: Field 'version' has not been initialized.

This happens on the flutter website

**Comment by parlough** [comment]
> Error compiling to JavaScript: LateInitializationError: Field 'version' has not been initialized.
> 
> This happens on the flutter website

Thanks for the report @tunjineo. I'm tracking the issue in https://github.com/dart-lang/dart-pad/issues/2494 and have a fix prepared at https://github.com/dart-lang/dart-pad/pull/2495.
